src/ripple/app/tests/TxQ_test.cpp

```diff
@@ -530,7 +613,919 @@ class TxQ_test : public beast::unit_test::suite
         // Alice's queued transaction failed in TxQ::accept
         // with tefPAST_SEQ
         checkMetrics(env, 0, 8, 0, 4, 256, 500);
+    }
+
+    void testMultiTxnPerAccount()
+    {
+        using namespace jtx;
+
+        Env env(*this, makeConfig({ { "minimum_txn_in_ledger_standalone", "3" } }),
+            features(featureFeeEscalation));
+
+        auto alice = Account("alice");
+        auto bob = Account("bob");
+        auto charlie = Account("charlie");
+        auto daria = Account("daria");
+
+        auto queued = ter(terQUEUED);
+
+        expect(env.current()->fees().base == 10);
+
+        auto lastMedian = 500;
+        checkMetrics(env, 0, boost::none, 0, 3, 256, lastMedian);
+
+        // Create several accounts while the fee is cheap so they all apply.
+        env.fund(XRP(50000), noripple(alice, bob, charlie, daria));
+        checkMetrics(env, 0, boost::none, 4, 3, 256, lastMedian);
+
+        // Alice - price starts exploding: held
+        env(noop(alice), queued);
+        checkMetrics(env, 1, boost::none, 4, 3, 256, lastMedian);
+
+        // Alice - try to queue a second transaction, but leave a gap
+        env(noop(alice), seq(env.seq(alice) + 2), fee(100),
+            ter(terPRE_SEQ));
+        checkMetrics(env, 1, boost::none, 4, 3, 256, lastMedian);
+
+        // Alice - queue a second transaction. Yay.
+        env(noop(alice), seq(env.seq(alice) + 1), fee(13),
+            queued);
+        checkMetrics(env, 2, boost::none, 4, 3, 256, lastMedian);
+
+        // Alice - queue a third transaction. Yay.
+        env(noop(alice), seq(env.seq(alice) + 2), fee(17),
+            queued);
+        checkMetrics(env, 3, boost::none, 4, 3, 256, lastMedian);
+
+        // Bob - queue a transaction
+        env(noop(bob), queued);
+        checkMetrics(env, 4, boost::none, 4, 3, 256, lastMedian);
+
+        // Bob - queue a second transaction
+        env(noop(bob), seq(env.seq(bob) + 1), fee(50),
+            queued);
+        checkMetrics(env, 5, boost::none, 4, 3, 256, lastMedian);
+
+        // Charlie - queue a transaction, with a higher fee
+        // than default
+        env(noop(charlie), fee(15), queued);
+        checkMetrics(env, 6, boost::none, 4, 3, 256, lastMedian);
+
+        auto aliceSeq = env.seq(alice);
+        auto bobSeq = env.seq(bob);
+        auto charlieSeq = env.seq(charlie);
+
+        env.close();
+        // Verify that all of but one of the queued transactions
+        // got applied.
+        lastMedian = 500;
```

Comment:
Every line of code is sacred
Every line of code is great
If a line of code is wasted
God gets quite irate